packages/components/addon/components/hds/dialog/header.hbs

```diff
@@ -0,0 +1,11 @@
+<div class="hds-dialog__header" ...attributes>
+  {{#if @icon}}
+    <FlightIcon class="hds-dialog__icon" @name={{@icon}} @size="24" @isInlineBlock={{false}} />
+  {{/if}}
+  <span class="hds-dialog__title" id={{@id}}>
+    {{yield}}
+  </span>
+  <button class="hds-dialog__dismiss" type="button" aria-label="Dismiss" {{on "click" @closeDialog}}>
+    <FlightIcon @name="x" @size="16" @isInlineBlock={{false}} />
+  </button>
```

Comment:
We did decide to decouple the `Dismissbutton` in Figma to it's own utility component. I'll work with @cveigt to get that handled.

Reply:
@jorytindall **probably** – and probably you already know it :) – the best and safest way to do it is to move the component already existing in the `Alert` and `Toast` component to its own page, so that possible designs already using the `Alert/Toast` components don't get the instance detached/reset.

Reply:
Following up on this, I've moved the `_dismissButton` out of the Alert and Toast components in [this branch](https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/branch/pNGhFl305jijdRTqrxBdpX/HDS-Product---Components?node-id=4452%3A7661). It's now on the **Utilities** page, I'll go ahead and merge the Figma branch for now!

Reply:
Based on the thread above, shall I mimic this in the code as well? To have a `DismissButton` 'utility' component shared between `Alert` and `Dialog`?

Reply:
That makes sense to me if they can easily share. It was literally just a copy/paste in Figma, so the visual and states are all exactly the same.

Reply:
> Based on the thread above, shall I mimic this in the code as well? To have a `DismissButton` 'utility' component shared between `Alert` and `Dialog`?

Yes, for me. My suggestion is to open a new PR (or I can do it), extract it from the `Alert/Toast` component and then when it's merged "consume" it in the `Dialog`.

Reply:
I added [a commit](https://github.com/hashicorp/design-system/pull/631/commits/879ba447841d1a9df0f76d1896d8e79cd4eabbfd) to abstract the `DismissButton` as a utility component shared by Alert/Toast and Dialog. Hopefully, it's small enough to be reviewed as part of this work, but if not let me know and I can open a separate PR.

Reply:
Works for me. Already reviewed and add a comment (what we discussed in the call)